src/planner.c

```diff
@@ -422,6 +435,949 @@ timescaledb_set_rel_pathlist_query(PlannerInfo *root, RelOptInfo *rel, Index rti
 	return;
 }
 
+#if PG12
+
+static void
+set_rel_pathlist(PlannerInfo *root, RelOptInfo *rel, Index rti, RangeTblEntry *rte);
+
+
+/* copied from allpaths.c */
```

Comment:
Will it be good idea to move all copied code into a separate file? The rational is that the file is too big already and it will be easier to distinguish between the code origin.